Week8AssignmentNew/ConsoleApp5/HomePage1.cs

```diff
@@ -0,0 +1,260 @@
+using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+
+namespace Automation2
+{
+    public class HomePage1
+    {
+        IWebDriver driver;
+        public HomePage1(IWebDriver d)
+        {
+            driver = d;
+        }
+
+        //Find the HomePage field
+        By textBox = By.XPath("//input[@title='Search for products, brands and more']");
+
+        //Setting the minimun range
+        By minDropdownElement = By.XPath("//span[text()='Price']/ancestor::section/*/following-sibling::*//select[contains(.,'Min')]");
+
+        //Setting header
+        By header = By.XPath("//body/div[@id='container']/div/div[1]/div[1]");
+        By maxDropdownElement = By.XPath("//option[@value='Min']/../../following-sibling::div[not(contains(.,'to'))]/select");
+
+        //Setting filter
+        By availability = By.XPath("//div[contains(text(),'Availability')]");
+        By excludeOutOfStockFilter = By.XPath("//div[contains(text(),'Exclude')]");
+
+        //Scrolling through the pages
+        By pageNumber = By.XPath("//div[@class='_2zg3yZ']/span[contains(text(),'Page')]");
+
+        //Identifying the product
+        String productName = "(//a[contains(@href,'apple-iphone')]/div[2]/div/div[contains(text(),'Apple iPhone')])[{0}]";
+
+        //Identifying the product price
+        String productPrice = "(//a[contains(@href,'apple-iphone')]/div[2]/div[2]/div[1]/div/div[1][contains(text(),'₹')])[{0}]";
+
+        //Storing the count of product displayed for a search
+        By searchResultCount = By.XPath("//span[contains(text(),'results')][1]");
+
+        //Displaying the count of the matching items present
+        By totalItems = By.XPath("//span[contains(text(),'Showing')]");
+
+        //Item unavailable
+        String unavailableItem = "(//img[contains(@alt,'Apple')])[{0}]/ancestor::div/following-sibling::div/span";
+
+        //Available item
+        String availableitem = "(//img[contains(@alt,'Apple')])[{0}]";
+
+        //List of all the products 
+        //By TextProd = By.XPath("(//a[contains(@href,'apple-iphone-6')]/div[2]/div/div[1][contains(text(),'Apple iPhone ')])[j]");
+
+        //Search result 
+        By searchResult = By.XPath("//div[contains(text(),'My Cart')]");
+
+        //Next button
+        By nextButton = By.XPath("//span[contains(text(),'Next')]");
+
+        //Total cost of the products
+        By totalCost = By.XPath("//span[contains(text(),'Price details')]/../div/div/span[contains(text(),'₹')]");
+
+        //Display the product name        
+        string addedProductsName = "(//div[contains(@class,'PaJLWc')])[{0}]/div[1]/div[1]/a[contains(text(),'Apple iPhone')]";
+
+        //Display the product cost
+        string addedProductsCost = "((//div[contains(@class,'PaJLWc')])[{0}]/div[1]/span[contains(text(),'₹')])";
+
+        String nameOfProduct;
+        String priceOfProduct;
+        int itemPrice;
+        string NumberOfItems;
+        Boolean boolResult;
+        public int itemsCount;
```

Comment:
Such variables should only be declared within the methods where they are needed

Reply:
i have removed them.